core/src/main/scala/org/bitcoins/core/wallet/utxo/StorageLocationTag.scala

```diff
@@ -0,0 +1,20 @@
+package org.bitcoins.core.wallet.utxo
+
+sealed abstract class StorageLocationTag extends AddressTag {
```

Comment:
So I think this is a problem -- if we want this to be extensible it seems that we actually _don't_ want this to be `sealed`. I.e. my business use case tomorrow is facilitating `LukewarmStorage` or whatever. How do I add this without coming in and modifying source code? 

If anything we maybe want to do this: 

```scala
sealed abstract class StorageLocationTag extends AddressTag {
  val typeName: String = "StorageLocationTag"
}

sealed abstract class DefaultStorageTags extends StorageLocationTag

/** Users can extend this to create their own storage tags */
abstract class UserDefinedStorageTags extends StorageLocationTag


object DefaultStorageTags extends AddressTagFactory[DefaultStorageTags] {

  /** Keys stored on a computer connected to the internet */
  final case object HotStorage extends DefaultStorageTags

  /** Keys stored on a hardware wallet or other offline device */
  final case object ColdStorage extends DefaultStorageTags

  /** Keys stored on a hardware wallet or other offline device locked in a safe in a distant location */
  final case object DeepColdStorage extends DefaultStorageTags

  val all: Vector[DefaultStorageTags] =
    Vector(HotStorage, ColdStorage, DeepColdStorage)
}
```

Reply:
`StorageLocationTag` is an example use case of extending AddressTag, this commit should not be merged. Check PR description.